src/lib.rs

```diff
@@ -0,0 +1,171 @@
+extern crate bit_vec;
+extern crate num;
+extern crate rand;
+extern crate rustty;
+
+use std::collections::{HashSet};
+
+use bit_vec::BitVec;
+use num::bigint::BigInt;
+use rand::Rng;
+use rustty::ui::{Widget};
+use rustty::{CellAccessor};
+
+type Cell = (usize, usize);
+type CellSet = HashSet<Cell>;
+
+pub struct World {
+    height: usize,
+    width: usize,
+    rule: BitVec,
+    grid: CellSet,
+}
+
+impl World {
+    pub fn new((width, height): Cell, rule: BigInt) -> World {
+        let (_, bytes) = rule.to_bytes_be();
+        World {
+            height: height,
+            width: width,
+            rule: BitVec::from_bytes(&bytes),
+            grid: HashSet::with_capacity(height * width),
+        }
+    }
+
+    pub fn gen(&mut self) {
+        self.grid.clear();
+        for x in 1..self.width {
+            for y in 1..self.height {
+                if rand::thread_rng().gen_weighted_bool(30) {
+                    self.grid.insert((x, y));
+                }
+            }
+        }
+    }
+
+    fn get_present_state(&self, cell: &Cell) -> usize {
```

Comment:
Why not just call it `get_state()`? Present state seems redundant in this context.

Reply:
done